app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,11 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      taxable_records = order.line_items + order.shipments
+      taxable_records.each do |taxable_record|
+        taxable_record.update_column(:pre_tax_amount, taxable_record.discounted_amount.round(2))
+      end
```

Comment:
I'm not sure where the responsibility should live, but I was taken aback by an invoice generator updating records on the thing it is generating an invoice for.

It might not be feasible to fix now, but I does seem a bit strange to me.